Source/LumatoneController.h

```diff
@@ -0,0 +1,407 @@
+/*
+  ==============================================================================
+
+    LumatoneCommandManager.h
+    Created: 17 Apr 2021 10:17:31pm
+    Author:  Vincenzo
+
+  ==============================================================================
+*/
+
+#pragma once
+
+#include "KeyboardDataStructure.h"
+#include "TerpstraMidiDriver.h"
+#include "DeviceActivityMonitor.h"
+#include "FirmwareTransfer.h"
+
+//==============================================================================
+// Helper class for parsing and comparing (todo) firmware versions
+
+
+
+class LumatoneController :  protected TerpstraMidiDriver::Listener, 
+                            protected FirmwareTransfer::ProcessListener,
+                            private juce::Thread::Listener,
+                            private juce::Timer, 
+                            private juce::ChangeListener
+{
+public:
+
+    enum sysExSendingMode
+    {
+        liveEditor = 0,
+        offlineEditor = 1,
+        firmwareUpdate = 2
+    };
+
+public:
+
+    LumatoneController();
+    ~LumatoneController();
+
+    //============================================================================
+    // Methods to configure firmware communication parameters
+
+    sysExSendingMode getSysExSendingMode() const { return editingMode; }
+    void setSysExSendingMode(sysExSendingMode newMode);
+
+    FirmwareVersion getFirmwareVersion() const { return firmwareVersion; }
+
+    LumatoneFirmwareVersion getConfirmedVersion() const { return determinedVersion; }
+    
+    // Takes a generic firmware version and parses it into a recognized firmware version
+    void setFirmwareVersion(FirmwareVersion firmwareVersionIn);
+
+    // Takes a recognized firmware version
+    void setFirmwareVersion(LumatoneFirmwareVersion lumatoneVersion, bool parseVersion = true);
+
+    int getOctaveSize() const { return octaveSize; }
```

Comment:
Same here for variable name, "octaveBoardSize" or "sectionSize" (see my other comment).